esmvalcore/_recipe_checks.py

```diff
@@ -156,7 +157,16 @@ def data_availability(input_files, var, dirnames, filenames, log=True):
         # check time availability only for non-fx variables
         return
 
-    required_years = set(range(var['start_year'], var['end_year'] + 1))
+    start_year, end_year = _parse_period(var['timerange'])
+    if start_year is None and end_year is None:
+        start_year = int(var['timerange'].split('/')[0][0:4])
+        end_year = int(var['timerange'].split('/')[1][0:4])
+    if isinstance(start_year, str) and isinstance(end_year, str):
+        start_year = int(start_year[0:4])
+        end_year = int(end_year[0:4])
+    required_years = set(range(start_year, end_year + 1))
+    var['start_year'] = start_year
+    var['end_year'] = end_year
```

Comment:
Changing the contents of the `var` dictionary is a really unexpected side effect of a function that is supposed to check data availability. Could you think of a better place to add this information?

Reply:
A bunch of tests start breaking since I think they work directly with the check. Is it too bad to keep it here? Working with the start and end yeats is something that it's going to ideally be deprecated at some point anyway.